src/issue.rs

```diff
@@ -10,46 +10,37 @@ use std::collections::HashMap;
 const ISSUE_NEEDS_A_PROJECT_MESSAGE: &str =
 	"{1} needs to be attached to a project or it will be closed.";
 
-fn issue_actor_and_project(
+async fn issue_actor_and_project(
 	issue: &github::Issue,
 	github_bot: &GithubBot,
 ) -> Result<Option<(github::User, github::Project)>> {
 	let repo = &issue.repository;
 	let issue_number = issue.number.context(error::MissingData)?;
-	github_bot
+	let actor_and_card = github_bot
 		.issue_events(&repo.name, issue_number)
-		.map(|issue_events| {
-			issue_events
-				.iter()
-				.sorted_by_key(|ie| ie.created_at)
-				.rev()
-				.find(|issue_event| {
-					issue_event.event == Some(github::Event::AddedToProject)
-						|| issue_event.event
-							== Some(github::Event::RemovedFromProject)
-				})
-				.and_then(|issue_event| {
-					if issue_event.event == Some(github::Event::AddedToProject)
-					{
-						issue_event.project_card.as_ref().and_then(|card| {
-							card.project_url.as_ref().map(|project_url| {
-								(issue_event.actor.clone(), project_url)
-							})
-						})
-					} else {
-						None
-					}
-				})
-				.and_then(|(actor, project_url)| {
-					github_bot
-						.get(project_url)
-						.ok()
-						.map(|project| (actor, project))
-				})
-		})
+		.await?
+		.into_iter()
+		.sorted_by_key(|ie| ie.created_at)
+		.rev()
+		.find(|ie| ie.event == Some(github::Event::AddedToProject))
```

Comment:
This alone won't tell us whether the issue is currently attached to a project, which is what the lines you've deleted above are for.